ecommerce/serializers.py

```diff
@@ -116,3 +151,79 @@ def get_coupons(self, instance):
     class Meta:
         fields = ["items", "coupons"]
         model = models.Basket
+
+
+class CouponPaymentSerializer(serializers.ModelSerializer):
+    """ Serializer for coupon payments """
+
+    class Meta:
+        fields = "__all__"
+        model = models.CouponPayment
+
+
+class CouponPaymentVersionSerializer(serializers.ModelSerializer):
+    """ Serializer for coupon payment versions """
+
+    payment = CouponPaymentSerializer()
+
+    class Meta:
+        fields = "__all__"
+        model = models.CouponPaymentVersion
+
+
+class BaseCouponSerializer(serializers.Serializer):
+    """ Base serializer for coupon creation data """
+
+    name = serializers.CharField(
+        max_length=256,
+        validators=[UniqueValidator(queryset=CouponPayment.objects.all())],
+    )
+    tag = serializers.CharField(max_length=256, allow_null=True, required=False)
+    amount = serializers.DecimalField(decimal_places=2, max_digits=20)
+    automatic = serializers.BooleanField(default=False)
+    activation_date = DateTimeTzField()
+    expiration_date = DateTimeTzField()
+    products = serializers.ListField(child=IntegerField())
```

Comment:
Suggest renaming this to `product_ids` since objects aren't being passed

Reply:
👍